Modules/Module_Import/Import/xASL_imp_DetermineSubjectStructure.m

```diff
@@ -112,37 +104,40 @@
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
 % EXAMPLE:        n/a
 % __________________________________
-% Copyright 2015-2021 ExploreASL
-
+% Copyright 2015-2024 ExploreASL
 
     %% Determine structure from temp data
 
     % Get subject/session list
     listSubjectsSessions = xASL_adm_GetFileList(x.modules.import.imPar.TempRoot,[],false,[],true);
     
+	% Initialize an empty list of sessions
+	listSessions = {};
+
     % Get subjects from list
     if numel(listSubjectsSessions)>0
         for iSubSes=1:numel(listSubjectsSessions)
             % Get current subject/session name
             curSubSes = listSubjectsSessions{iSubSes};
             % Determine subject name
-            if ~isempty(regexp(curSubSes,'_','all')) && ~(numel(regexp(curSubSes,'_','all'))>1)
-                % Multi-session notation
-                x.SUBJECTS{iSubSes} = curSubSes(1:regexp(curSubSes,'_')-1);
-            elseif isempty(regexp(curSubSes,'_','all'))
+			indexSeparator = regexp(curSubSes, '_', 'all');
+            if isempty(indexSeparator)
                 % Single-session exceptions
                 x.SUBJECTS{iSubSes} = curSubSes;
-            else
-                warning('It was not possible to determine the subject name from the temp data...');
-                if numel(regexp(curSubSes,'_','all'))>1
-                    warning('Multiple underscores in subject/session name...');
-                end
+				listSessions{iSubSes} = '';
+			elseif numel(indexSeparator) == 1
+                % Multi-session notation
+                x.SUBJECTS{iSubSes} = curSubSes(1:indexSeparator-1);
+				listSessions{iSubSes} = curSubSes(indexSeparator+1:end);
+			else
+				% Multiple separators
+                warning('It was not possible to determine the subject name from the temp data because there are multiple underscores in subject/session name');
```

Comment:
Not specific enough, I would make it:

warning('It was not possible to determine the subject (or session) name from temporary data, note that underscores in subject or session values are illegal. Please check your data with the BIDS validator.'

Reply:
I changed the warning. But removed the last sentence. This is about DICOM data and DCM2NII - we can't run that through the BIDS validator.

Reply:
Ah sorry I misunderstood. My points were also:

1. User doesn't know what temp data is. It doesn't exist to the user (and shouldn't, it is just a dirty in between-stage to make our life easy).
2. The problem probably doesn't lie in the temp data stage but in the initial naming of the folders.
3. The initial naming of the folders is often taken from standard scanner output.

So I would suggest then something like:

warning('In BIDS no values (==names) can have hyphens (-) or underscores (_) because these are used as separators. Removing them in the final BIDS values/names')